python/mlc_chat/interface/bench.py

```diff
@@ -0,0 +1,28 @@
+"""Python entrypoint of benchmark."""
+from typing import Optional
+
+from mlc_chat.chat_module import ChatConfig, ChatModule
+
+from .compiler_flags import ModelConfigOverride
+
+
+def bench(  # pylint: disable=too-many-arguments
+    model: str,
+    prompt: str,
+    device: str,
+    opt: str,
+    overrides: ModelConfigOverride,
+    generate_length: int,
+    model_lib_path: Optional[str],
+):
+    """run the benchmarking"""
+    # Set up chat config and generate config
+    config = ChatConfig(opt=opt)
+    # Apply overrides
+    config = overrides.apply(config)
```

Comment:
Unfortunately it's a bit more complicated here. `ModelConfigOverride` is designed to only apply to `ModelConfig`, such as `LlamaConfig`, etc., rather than ChatConfig. We will have to add most of those flags to argparse :(((

Reply:
of those flags, the ones below are important:
- conv_template
- tensor_parallel_shards
- context_window_size
- sliding_window_size
- prefill_chunk_size
- attention_sink_size
- max_batch_size
- opt